src/util/system.cpp

```diff
@@ -899,25 +782,31 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
         bool emptyIncludeConf;
         {
             LOCK(cs_args);
-            emptyIncludeConf = m_override_args.count("-includeconf") == 0;
+            const std::vector<util::SettingsValue>* const includes = util::FindKey(m_settings.command_line_options, "includeconf");
+            emptyIncludeConf = !(includes && util::SettingsSpan(*includes).last_negated());
         }
         if (emptyIncludeConf) {
             std::string chain_id = GetChainName();
-            std::vector<std::string> includeconf(GetArgs("-includeconf"));
-            {
-                // We haven't set m_network yet (that happens in SelectParams()), so manually check
-                // for network.includeconf args.
-                std::vector<std::string> includeconf_net(GetArgs(std::string("-") + chain_id + ".includeconf"));
-                includeconf.insert(includeconf.end(), includeconf_net.begin(), includeconf_net.end());
-            }
+            std::vector<std::string> includeconf;
```

Comment:
These variable names (which are unchanged by your PR) aren't very helpful. I'd suggest the following renames for a later PR:

`includeconf` -> `conf_file_names`
`to_include` -> `conf_file_name`
`include_conf` -> `conf_file_stream`